lib/resourceutil/util.go

```diff
@@ -0,0 +1,97 @@
+package resourceutil
+
+import (
+	"sort"
+	"time"
+
+	"github.com/go-logr/logr"
+	"github.com/syntasso/kratix/lib/hash"
+	"github.com/syntasso/kratix/lib/pipeline"
+	batchv1 "k8s.io/api/batch/v1"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	clusterv1 "sigs.k8s.io/cluster-api/api/v1beta1"
+)
+
+const PipelineCompletedCondition = clusterv1.ConditionType("PipelineCompleted")
+
+func HasPipelineCompletedCondition(obj *unstructured.Unstructured) bool {
+	return HasCondition(obj, PipelineCompletedCondition)
+}
+
+func LastPipelineCompleted(obj *unstructured.Unstructured) bool {
+	condition := GetCondition(obj, PipelineCompletedCondition)
+	return condition != nil && condition.Status == v1.ConditionTrue
+}
+
+func MarkPipelineAsRunning(logger logr.Logger, obj *unstructured.Unstructured) {
+	SetCondition(obj, &clusterv1.Condition{
+		Type:               PipelineCompletedCondition,
+		Status:             v1.ConditionFalse,
+		Message:            "Pipeline has not completed",
+		Reason:             "PipelineNotCompleted",
+		LastTransitionTime: metav1.NewTime(time.Now()),
+	})
+	logger.Info("set conditions", "condition", PipelineCompletedCondition, "value", v1.ConditionFalse)
+}
+
+func MarkPipelineAsCompleted(logger logr.Logger, obj *unstructured.Unstructured) {
+	SetCondition(obj, &clusterv1.Condition{
+		Type:               PipelineCompletedCondition,
+		Status:             v1.ConditionTrue,
+		Message:            "Pipeline completed",
+		Reason:             "PipelineExecutedSuccessfully",
+		LastTransitionTime: metav1.NewTime(time.Now()),
+	})
+	logger.Info("set conditions", "condition", PipelineCompletedCondition, "value", v1.ConditionTrue)
+}
+
+func IsAnUpdate(logger logr.Logger, rr *unstructured.Unstructured, jobs []batchv1.Job) bool {
+	if len(jobs) == 0 {
+		return false
+	}
+
+	// sort the pipepeineJobs by creation date
+	sort.Slice(jobs, func(i, j int) bool {
+		t1 := jobs[i].GetCreationTimestamp().Time
+		t2 := jobs[j].GetCreationTimestamp().Time
+		return t1.Before(t2)
+	})
+
+	lastPipelineJob := jobs[len(jobs)-1]
+
+	lastRequestHash := lastPipelineJob.GetLabels()[pipeline.KratixResourceHashLabel]
+	currentRequestHash, err := hash.ComputeHash(rr)
+	if err != nil {
+		logger.Info("Cannot determine if the request is an update. Requeueing", "reason", err.Error())
+		return false
+	}
+
+	return lastRequestHash != currentRequestHash
+}
```

Comment:
should this live here? It feels like really heavy business logic when compared to the other functions that are in this package. Not a hill I'm going to die on, but this func stands out compared to the rest

Reply:
i don't really know if i need to sort the jobs, just playing it safe here; i can see moving this out of here, but also don't care too much 👀 apart from the sort, its really just comparing the hashes.